index.js

```diff
@@ -2,34 +2,64 @@ const Entity = require('siren-parser');
 const Action = require('siren-parser/dist/Action');
 const Link = require('siren-parser/dist/Link');
 
+/**
+ * siren-parser wrapper that supports following links and performing actions
+ */
 class Client {
   /**
    * performs an async HTTP request
    * returns object { body, contentType }
    *  body: response body
    *  contentType: Content-Type response header
    *
-   * @callback request
-   * @param {string} uri (required) uri of the resource
+   * @typedef response
+   * @property body response body
+   * @property contentType Content-Type response header
+   *
+   * @callback requestFn
+   * @param {string} href (required) href of the resource
    * @param {string} method (required) HTTP request verb ('GET', 'POST', 'PUT', 'PATCH', 'DELETE')
-   * @param {object} fieldValues (optional) field names and values for the request
-   * @return {object} body: response body, contentType: Content-Type response header
+   * @param {object} [fieldValues] (optional) field names and values for the request
+   * @return {response} body: response body, contentType: Content-Type response header
    */
 
   /**
    * Setup the fluent siren client
    *
-   * @param {request} requestFn performs an async HTTP request
+   * @typedef options
+   * @type {object}
+   * @property {String|String[]} [originWhitelist] (optional) Whitelist of allowed origins (e.g. 'https://example.com')
+   *
+   * @param {requestFn} requestFn (required) performs an async HTTP request
+   * @param {options} [options] (optional) configuration options for the client
    */
-  constructor(requestFn) {
+  constructor(requestFn, options = {}) {
     this._requestFn = requestFn;
+    this._options = options;
   }
 
+  /**
+   * Starts a client by fetching the provided href
+   * @param href
+   * @returns {Promise<Entity>}
+   */
   async start(href) {
     return this.perform(href, 'GET');
   }
 
+  /**
+   * Performs a request
+   * @param href
+   * @param method
+   * @param [fieldValues]
+   * @returns {Promise<Entity>}
+   */
   async perform(href, method, fieldValues) {
+    verifyOrigin(href, this._options.originWhitelist);
+    return this._perform(href, method, fieldValues);
+  }
+
+  async _perform(href, method, fieldValues) {
```

Comment:
is this ok as is or should it be made completely private (a method outside the class)?

Reply:
another similar question: should we make attachClient() below private or prefix it with underscore? Do we expect it to be called by the users of the class?

Reply:
I think attachClient should stay public, as it does support a use case where you want to attach an existing entity to this Client instance. 

I think `_perform` could be fully private however, as we really shouldnt be circumventing this sort of thing. If you need to circumvent this, make a new client with no origin whitelist.

Reply:
sounds good to me. Will work on making it private.